lib/rmagick_internal.rb

```diff
@@ -8,6 +8,18 @@
 #               to the classes.
 #==============================================================================
 
+if RUBY_PLATFORM =~ /mingw/i
+  begin
+    require 'ruby_installer'
+    ENV['PATH'].split(File::PATH_SEPARATOR).each do |path|
+      if path =~ /ImageMagick/i
+        RubyInstaller::Runtime.add_dll_directory(path)
+      end
+    end
+  rescue
```

Comment:
Are there specific errors we expect to be catching here? I wonder if we can do without the `begin`/`rescue`.

Reply:
I think that the exceptions will be raised by two pattern at least.

1. `LoadError`
`ruby_installer` is provided by [RubyInstaller](https://rubyinstaller.org).

MinGW environment might be constructed without [RubyInstaller](https://rubyinstaller.org) by user self compiled using [MinGW](http://www.mingw.org).

If so, the user environment can't load `ruby_installer`.

2. `NoMemoryError`

[RubyInstaller](https://rubyinstaller.org) was changed its package since Ruby 2.4 by changing mentener.

`ruby_installer` might not have `add_dll_directory` method in older Ruby package.

(I didn't research it... sorry)

Reply:
I see, thanks for the explanation. If there are users who might not have used RubyInstaller, will they also see this error? If so, how hard would it be to fix the issue without using RubyInstaller? I was looking at [their source](https://github.com/oneclick/rubyinstaller2/blob/1dfa3f5e407345f11936da21c578e426f76d55a8/lib/ruby_installer/build/dll_directory.rb), but not sure what parts are important, or if it's worth implementing here.

Reply:
The [docs also mention](
https://github.com/oneclick/rubyinstaller2/wiki/For-gem-developers#-dll-loading) possibly setting the `RUBY_DLL_PATH` env variable.

Reply:
Actually, looking at `ruby-pg`, they're also using the RubyInstaller approach, so I guess that's fine. But in terms of the `rescue` it looks like they're only worried about `LoadError`, so maybe we should just catch that?

Also, they're looking for both `mswin` and `mingw`, so we may want to do that as well.

Reply:
https://github.com/ged/ruby-pg/blob/33656e4aefdd543364ea154c0309956e3c63fd2a/lib/pg.rb#L14

Reply:
Thank you. I modified to catch only LoadError.